src/example_documentation.cpp

```diff
@@ -0,0 +1,128 @@
+// DESCRIPTION: Appropriate license header at the top, e.g.,
+/*
+* Copyright (c) 2018, NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+// DESCRIPTION: A brief description of the overall purpose and contents of this file
+/** ---------------------------------------------------------------------------*
+* @brief Example code documentation for libgdf.
+*
+* This file provides examples of how source files, classes, functions, and variables
+* should be documented in libgdf.
+* ---------------------------------------------------------------------------**/
+
+
+// DESCRIPTION: A brief description of the purpose and functionality of the class 
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief  This class serves as an example of how classes in libgdf should 
```

Comment:
Please give an example of using both @brief and detailed documentation in a block. In doxygen, the @brief will end up in lists and summaries, whereas the detailed description goes on the full docs page for the function / class. Here's an example. Note the lines after the @brief are the detailed description.

```/** ---------------------------------------------------------------------------*
 * @brief Get amounts of free and total memory managed by a manager associated with the stream.
 * 
 * Returns in *free and *total, respectively, the free and total amount of memory available 
 * for allocation by the device in bytes.
 * 
 * @param[out] freeSize The free memory in bytes available to the manager associated with stream
 * @param[out] totalSize The total memory managed by the manager associated with stream
 * @param[in] stream 
 * @return rmmError_t RMM_SUCCESS, or RMM_ERROR_NOT_INITIALIZED if rmmInitialize has not
 *                    been called, or RMM_ERROR_CUDA_ERROR on any CUDA error
 * ---------------------------------------------------------------------------**/
rmmError_t rmmGetInfo(size_t *freeSize, size_t *totalSize, cudaStream_t stream);```